### PR TITLE
Return lists of inactive localities and countries

### PR DESCRIPTION
**Pull request by konstyefremov**

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Controllers/LocationController.cs [hunk 1]
```diff
@@ -70,6 +70,27 @@ public async Task<IActionResult> GetSlimLocationDescription(string htId)
...
+            => Ok(await _locationService.GetInactiveCountries(htIds));
+
+
+        /// <summary>
+        /// Gets inactive localities
+        /// </summary>
+        /// <param name="htIds">List of htIds of localities to check</param>
+        /// <returns>List of inactive localities</returns>
+        [HttpPost("inactive-localities")]
+        public async Task<IActionResult> GetInactiveLocalities(List<string> htIds)
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Api/Controllers/LocationController.cs:90]: There is must be list size restriction

File: HappyTravel.Nakijin.Api/Services/LocationService.cs [hunk 2]
```diff
@@ -69,6 +69,30 @@ select new
...
+            
+            return await _context.Countries
+                .Where(c => !c.IsActive)
+                .Where(c => parsedIds.Contains(c.Id))
+                .Select(c => HtId.Create(MapperLocationTypes.Country, c.Id))
+                .ToListAsync();
+        }
+        
+        
+        public async Task<List<string>> GetInactiveLocalities(IEnumerable<string> htIds)
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Api/Services/LocationService.cs:84]: Some of the inactive localities could be linked to the active localities as synonyms

File: HappyTravel.Nakijin.Core/HtId.cs [hunk 3]
```diff
@@ -27,7 +29,29 @@ public static Result<(MapperLocationTypes Type, int Id)> Parse(string htId)
...
+                return Result.Failure<int>(error);
+
+            if (value.Type != type)
+                return Result.Failure<int>($"{value.Type} is not supported");
+
+            return value.Id;
+        }
+
+
+        public static List<int> Parse(MapperLocationTypes type, IEnumerable<string> htIds)
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Core/HtId.cs:46]: All items of the list must be parsed correctly. If not then Result.Failure should be returned

File: HappyTravel.Nakijin.Api/Controllers/LocalityController.cs [hunk 4]
```diff
@@ -37,8 +39,26 @@ public async Task<IActionResult> Get(string htId)
...
+
+
+        /// <summary>
+        /// Get localities and synonyms matches
+        /// </summary>
+        /// <param name="htIds">Ids of localities to check</param>
+        /// <returns></returns>
+        [ProducesResponseType(typeof(List<LocalityAndSynonymMatch>), (int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [HttpPost("localities-and-synonyms-matches")]
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Api/Controllers/LocalityController.cs:51]: Maybe localities/synonyms/match ?

File: HappyTravel.Nakijin.Api/Controllers/LocationController.cs [hunk 5]
```diff
@@ -70,6 +70,42 @@ public async Task<IActionResult> GetSlimLocationDescription(string htId)
...
         
         
+        /// <summary>
+        /// Gets inactive countries from list of htIds
+        /// </summary>
+        /// <param name="htIds">List of htIds to check</param>
+        /// <returns>List of inactive countries</returns>
+        [ProducesResponseType(typeof(List<string>), (int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [HttpPost("inactive-countries")]
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Api/Controllers/LocationController.cs:80]: countries/deactivated ?

File: HappyTravel.Nakijin.Api/Controllers/LocalityController.cs (hunk 4 above)
Comment (konstyefremov) [reply HappyTravel.Nakijin.Api/Controllers/LocalityController.cs]: Agree, I like it.

File: HappyTravel.Nakijin.Api/Controllers/LocationController.cs (hunk 5 above)
Comment (konstyefremov) [reply HappyTravel.Nakijin.Api/Controllers/LocationController.cs]: Agree, I like it.

**Review by oleg-n-k — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Controllers/LocationController.cs [hunk 6]
```diff
@@ -70,6 +70,42 @@ public async Task<IActionResult> GetSlimLocationDescription(string htId)
...
         
+        /// <summary>
+        /// Gets inactive countries from list of htIds
+        /// </summary>
+        /// <param name="htIds">List of htIds to check</param>
+        /// <returns>List of inactive countries</returns>
+        [ProducesResponseType(typeof(List<string>), (int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [HttpPost("countries/deactivated")]
+        public async Task<IActionResult> GetInactiveCountries(List<string> htIds)
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Api/Controllers/LocationController.cs:81]: Also, the method could be renamed to GetDeactivatedCountries

**Review by mingazhev — APPROVED** [review]

**Pull request merged**